x/auth/keeper/keeper.go

```diff
@@ -233,3 +245,48 @@ func (ak AccountKeeper) UnmarshalAccount(bz []byte) (types.AccountI, error) {
 
 // GetCodec return codec.Codec object used by the keeper
 func (ak AccountKeeper) GetCodec() codec.BinaryCodec { return ak.cdc }
+
+// GetAddressCdc returns address.Codec object used by the keeper
+func (ak AccountKeeper) GetAddressCdC() address.Codec { return ak.addressCdC }
+
+// add getter for bech32Prefix
+func (ak AccountKeeper) GetBech32Prefix() (string, error) {
+	bech32Codec, ok := ak.addressCdC.(bech32Codec)
+	if !ok {
+		return "", errors.New("unable cast addressCdC to bech32Codec")
+	}
+
+	return bech32Codec.bech32Prefix, nil
+}
+
+func newBech32Codec(prefix string) bech32Codec {
+	return bech32Codec{prefix}
+}
+
+// AddressStringToBytes encodes text to bytes
+func (bc bech32Codec) AddressStringToBytes(text string) ([]byte, error) {
+	hrp, bz, err := bech32.DecodeAndConvert(text)
+	if err != nil {
+		return nil, err
+	}
+
+	if hrp != bc.bech32Prefix {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrLogic, "hrp does not match bech32Prefix")
+	}
+
+	if err := address.VerifyFormat(bz); err != nil {
```

Comment:
The application can have it's own verifier. See `types.VerifyAddressFormat`.

Reply:
Hmm, I believe the rationale behind using `VerifyFormat` was to remove dependency on globals.

Do you have any ideas how to not use globals here? Pass an optional address verifier to the AccountKeeper constructor?

Or maybe we just use `types.VerifyAddressFormat`, and create another issue to think about this specifically.

Reply:
> Do you have any ideas how to not use globals here? Pass an optional address verifier to the AccountKeeper constructor?

I like this idea.

Reply:
remove verifyFormat